nix/shell.nix

```diff
@@ -24,6 +24,8 @@ in mkShell {
       clojure maven watchman
       # other nice to have stuff
       yarn nodejs python310 maestro
+      # Required for /scripts/compress_image.sh script
+      imagemagick
```

Comment:
also this would have to go in `overlays.nix` 
I'm AFK today but can check this PR out later

Reply:
Thank you for checking the PR. Please feel free to respond tomorrow.

> also this would have to go in overlays.nix

Please can you elaborate why?

Reply:
> [nix/overlay.nix](https://github.com/status-im/status-mobile/blob/develop/nix/overlay.nix) - Overrides for nixpkgs, custom packages

Reply:
Ah correct @Parveshdhull : overlays.nix is only if we are changing or overriding any version.
My bad.
This looks good.